pkg/iac/terraform/state/backend/http_reader_test.go

```diff
@@ -15,10 +19,11 @@ func TestNewHTTPReader(t *testing.T) {
 		options *Options
 	}
 	tests := []struct {
-		name    string
-		args    args
-		wantURL string
-		wantErr error
+		name       string
+		args       args
+		wantErr    error
+		httpClient func() HttpClient
```

Comment:
You can put `HttpClient` directly and in each case you can do `httpClient: func() HttpClient { return &http.Client{} }()`

This more homogenous with what we do in other tests

Reply:
Isn't it what's already done in this PR ?